linera-core/src/updater.rs

```diff
@@ -291,16 +293,38 @@ where
                     let certificates = local_storage
```

Comment:
(Not in this PR, but: We should probably check whether what the remote node asks for is reasonable, i.e. that all those blobs are actually used by this block?)

Reply:
I have indeed many issues with that line:
* We download the certificates, but after this is done, only the ones of highest height matter. Could we download only the ones of highest height.
* The downloaded information of the certificates contains a lot of stuff but we used only the `ChainId`/ `BlockHeight`.

The types are:
* : The `Certificate` type contains the HashedCertificateValue / Round / Vec<(ValidatorName,Signature)>
* The `HashedCertificateValue` contains the CertificateValue / CryptoHash.
* The `CertificateValue` contains executed block information or Timeout.

So, maybe some reorganization of the stored data could be done. But otherwise, yes maybe we could use the blocks and check whether they are indeed executed.

Reply:
Good point! I think `last_used_by` should just contain chain ID and height in addition to (or instead of?) the certificate hash. Then we wouldn't have to read them at all.